javalib/src/main/scala/java/util/regex/Pattern.scala

```diff
@@ -0,0 +1,203 @@
+      if ((flags & flag) == flag) {
+        assert(false, s"regex flag $flagName is not supported")
+      }
+    }
+
+    val options = cre2_opt_new()
+    notSupported(CANON_EQ, "CANON_EQ(canonical equivalences)")
+    cre2_opt_set_case_sensitive(options, flags & CASE_INSENSITIVE)
+    notSupported(COMMENTS, "COMMENTS")
+    cre2_opt_set_dot_nl(options, flags & DOTALL)
+    cre2_opt_set_literal(options, flags & LITERAL)
+    notSupported(UNICODE_CASE, "UNICODE_CASE")
+    notSupported(UNICODE_CHARACTER_CLASS, "UNICODE_CHARACTER_CLASS")
+    notSupported(UNIX_LINES, "UNIX_LINES")
+    // cre2_opt_set_one_line(false) is only available when limiting ourself to posix_syntax
+    // https://github.com/google/re2/blob/2017-03-01/re2/re2.h#L548
+    // regex flag MULTILINE cannot be disabled
+
+    cre2_opt_set_log_errors(options, 0)
+
+    val re2 = cre2_new(toCString(regex), regex.size, options)
+
+    val code = new cre2_error_code_t(cre2_error_code(re2))
+
+    if (code != CRE2_NO_ERROR) {
+      val errorPattern = {
+        val arg = cre2_string_t.stackalloc
+        cre2_error_arg(re2, arg)
+        arg.toString
+      }
+
+      // we try to find the index of the parsing error
+      // this could return the wrong index it only finds the first match
+      // see https://groups.google.com/forum/#!topic/re2-dev/rnvFZ9Ki8nk
+      val index =
+        if (code == CRE2_ERROR_TRAILING_BACKSLASH) regex.size - 1
+        else regex.indexOfSlice(errorPattern)
+
+      val reText = fromCString(cre2_error_string(re2))
+
+      val description =
+        code match {
+          case CRE2_ERROR_BAD_ESCAPE         => "Illegal/unsupported escape sequence"
+          case CRE2_ERROR_MISSING_PAREN      => "Missing parenthesis"
+          case CRE2_ERROR_TRAILING_BACKSLASH => "Trailing Backslash"
+          case CRE2_ERROR_MISSING_BRACKET    => "Unclosed character class"
+          case CRE2_ERROR_BAD_CHAR_RANGE     => "Illegal character range"
+          case CRE2_ERROR_BAD_CHAR_CLASS =>
+            "Illegal/unsupported character class"
+          case CRE2_ERROR_REPEAT_SIZE       => "Bad repetition argument"
+          case CRE2_ERROR_REPEAT_ARGUMENT   => "Dangling meta character '*'"
+          case CRE2_ERROR_REPEAT_OP         => "Bad repetition operator"
+          case CRE2_ERROR_BAD_PERL_OP       => "Bad perl operator"
+          case CRE2_ERROR_BAD_UTF8          => "Invalid UTF-8 in regexp"
+          case CRE2_ERROR_BAD_NAMED_CAPTURE => "Bad named capture group"
+          case CRE2_ERROR_PATTERN_TOO_LARGE =>
+            "Pattern too large (compilation failed)"
+          case CRE2_ERROR_INTERNAL => "Internal Error"
+          case _                   => reText
+        }
+
+      throw new PatternSyntaxException(
+        description,
+        regex,
+        index
+      )
+    }
+
+    new Pattern(
+      _pattern = regex,
+      _flags = flags,
+      _regex = re2,
+      options = options
```

Comment:
Why do you need options in pattern? They are never used anywhere in the Scala code oustide of `compile` method.